File: factorial/Fact.java
import javax.swing.*;

public class Fact {
	public static void main(String[] args) {
		final int inputNumber = Integer.parseInt(JOptionPane.showInputDialog(null, "Please type a number to calculate its factorial"));
		int cloneInputNumber = inputNumber;
		long result = 1;
		while (cloneInputNumber > 1){
			result *= cloneInputNumber--;
		}
		JOptionPane.showMessageDialog(null, "The result of " + inputNumber + "! is: " + result);
	}
}

File: factorial/fact.py
def fact(n):  
   if n == 1:  
       return n  
   else:  
       return n*fact(n-1)   
num = int(input("Input a number: "))  
if num < 0:  
   print("Sorry, you can't find factorial of negative value")  
elif num == 0:  
   print("The factorial of 0 is 1")  
else:  
   print("The factorial of"​,num,​"is"​,fact(num))  
